matching-notification-api/src/main/java/org/phenotips/matchingnotification/match/internal/DefaultPatientInMatch.java

```diff
@@ -498,21 +493,42 @@ private Patient getLocalPatient()
         }
     }
 
-    private void populateContactInfo(String href)
+    private ContactInfo populateContactInfo()
     {
-        this.href = null;
-        // if the patient is remote, we use whatever is passed by from DB
-        if (this.patient == null) {
-            this.href = href;
-            return;
+        if (this.patient != null && this.patient.getData("contact").size() > 0) {
+            PatientData<ContactInfo> data = this.patient.getData("contact");
+            if (data != null && data.size() > 0) {
+                return data.get(0);
+            }
         }
-        PatientData<ContactInfo> data = this.patient.getData("contact");
-        if (data != null && data.size() > 0) {
-            ContactInfo contact = data.get(0);
-            if (contact != null) {
-                this.href = contact.getUrl();
+        return null;
+    }
+
+    private ContactInfo rebuildContactInfo(JSONObject contact, String oldMMEContactData)
```

Comment:
if this is called via line 193 where we may have a new version of a patient (via `this.patient = getLocalPatient();`) we need to get contact info from patient as well, as it is higher priority than stored data.

I.e. the code should be something like

```
if (patient != null) {
   this.populateContactInfo()
} else if (contact != null) {
...
}
```